src/FloorSubsystem/FloorDataGramCommunicator.java

```diff
@@ -42,11 +38,12 @@ public FloorDataGramCommunicator() {
 
 	/**
 	 * Sends a message contained in a byte array using a datagram packet and
-	 * datagram socket to the scheduler
+	 * datagram socket to the scheduler. Also receives the same message from the
+	 * scheduler
 	 * 
 	 * @param message
 	 */
-	public void send(byte[] message) {
+	public void sendAndReceive(byte[] message) {
 
```

Comment:
If floor always expects a response after sending a message, then I agree, though this might be annoying to unit test

Reply:
I can split this into two methods